reposerver/repository/repository.go

```diff
@@ -2851,6 +2884,10 @@ func (s *Service) UpdateRevisionForPaths(_ context.Context, request *apiclient.U
 	}
 	defer io.Close(closer)
 
+	if err := s.fetch(gitClient, []string{syncedRevision, revision}); err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to fetch git repo %s with revisions (%s, %s): %v", repo.Repo, syncedRevision, revision, err)
+	}
```

Comment:
How would locking above work with both revisions being used here? Can we lock for multiple revisions?

Reply:
Should we take a lock to fetch from the remote? If we checkout we need to take a lock, but for fetch we don't need to lock.

Reply:
I don't know